src/background.js

```diff
@@ -35,63 +35,76 @@ const TIPS = {
   UPDATE_WEB: "update_web",
 };
 
-// Keywords for each tip type.
-const KEYWORDS = {
-  update: [
-    "2019",
-    "browser",
-    "download",
-    "fire",
-    "firefox",
-    "fox",
-    "free",
-    "get",
-    "install",
-    "installer",
-    "latest",
-    "mac",
-    "mozilla",
-    "new",
-    "newest",
-    "quantum",
-    "update",
-    "updates",
-    "version",
-    "windows",
-    "www.firefox.com",
-  ],
+// The search "documents" corresponding to each tip type.
+const DOCUMENTS = {
   clear: [
-    "cache",
-    "clear",
-    "cookie",
-    "cookies",
-    "delete",
-    "firefox",
-    "history",
-    "load",
-    "loading",
-    "loads",
-    "location",
-    "page",
+    "cache firefox",
+    "clear cache firefox",
+    "clear cache in firefox",
```

Comment:
For now this is ok, but long term we should probably add a list of stop words for common conjunctions that may be typed among the phrases. In other locales (like italian) those will appear a lot more often ("cancella cronologia Firefox" or "cancella la cronologia in/di Firefox").
Or make phrases actually just an array of words that **must all match** in a query, **in any order**. Then ["clear", "cache", "firefox"] would replace at least 3 entries here, or we'd not even need it, if we'd have ["firefox", "cache"] and ["clear", "cache"].

Reply:
That's good to think about, thanks. I feel like I just can't get this right.

Reply:
I'm tempted to forget this patch and implement your must-all-match and in-any-order idea...